engine/src/main/java/org/terasology/rendering/cameras/OpenVRStereoCamera.java

```diff
@@ -206,49 +211,68 @@ public void updateMatrices() {
         updateMatrices(activeFov);
     }
 
+    private void jomlMatrix4f(org.joml.Matrix4f matrixInput, org.terasology.math.geom.Matrix4f matrixOut) {
+        matrixOut.set(0, 0, matrixInput.m00());
+        matrixOut.set(0, 1, matrixInput.m10());
+        matrixOut.set(0, 2, matrixInput.m20());
+        matrixOut.set(0, 3, matrixInput.m30());
+        matrixOut.set(1, 0, matrixInput.m01());
+        matrixOut.set(1, 1, matrixInput.m11());
+        matrixOut.set(1, 2, matrixInput.m21());
+        matrixOut.set(1, 3, matrixInput.m31());
+        matrixOut.set(2, 0, matrixInput.m02());
+        matrixOut.set(2, 1, matrixInput.m12());
+        matrixOut.set(2, 2, matrixInput.m22());
+        matrixOut.set(2, 3, matrixInput.m32());
+        matrixOut.set(3, 0, matrixInput.m03());
+        matrixOut.set(3, 1, matrixInput.m13());
+        matrixOut.set(3, 2, matrixInput.m23());
+        matrixOut.set(3, 3, matrixInput.m33());
+    }
+
     @Override
     public void updateMatrices(float fov) {
         prevViewProjectionMatrix.set(viewProjectionMatrix);
 
-        // Nothing to do...
-        if (cachedPosition.equals(getPosition()) && cachedViewigDirection.equals(getViewingDirection())
-                && cachedZFar == zFar && cachedZNear == zNear) {
+        org.joml.Matrix4f leftEyeProjection = vrProvider.vrState.getEyeProjectionMatrix(0);
+        org.joml.Matrix4f rightEyeProjection = vrProvider.vrState.getEyeProjectionMatrix(1);
+        org.joml.Matrix4f leftEyePose = vrProvider.vrState.getEyePose(0);
+        org.joml.Matrix4f rightEyePose = vrProvider.vrState.getEyePose(1);
```

Comment:
JOML is needed here because VRProvider uses JOML matrices as outputs to this function. I initially wrote VRProvider to be a separate module and pulled it in. The upside is that VRProvider can easily be plucked out of Terasology and put elsewhere (and changes to VRProvider in Terasology can be brought out to the general repository from which it came, and be useful to others). I'm willing to compromise here if you guys insist.